### PR TITLE
TP2000-853: Add measures as defined tab to commodity detail

### PR DESCRIPTION
**Pull request by eadpearce**
# TP2000-853: Add measures as defined tab to commodity detail

## Why

* Users want to be able to see what measures are created for a given commodity on the commodity page without having to use the measures search

## What

* Adds a new tab to the commodity detail page for currently active measures on the given commodity
* Measures can be sorted by measure type, geographical area and start date

Screenshot 2023-04-27 at 16 44 50
Screenshot 2023-04-27 at 16 45 20

### CODE REVIEW

File: commodities/jinja2/includes/commodities/tabs/measures.jinja [hunk 1]
```diff
@@ -0,0 +1,53 @@
+{% from "components/sort_by.jinja" import sort_by %}
+
+{% set parent_code %}
+    {% if parent %}
+        <a class="govuk-link govuk-!-font-weight-bold" href="{{ url("commodity-ui-detail", kwargs={"sid": parent.sid}) }}">{{ parent.code }}</a>
```
Comment (a-gleeson) [inline commodities/jinja2/includes/commodities/tabs/measures.jinja:5]: Not sure if I am missing it but is this used?

File: commodities/tests/test_views.py [hunk 2]
```diff
@@ -183,3 +183,152 @@ def test_goods_nomenclature(valid_user_client, date_ranges):
     assert past_good.pk not in pks
     assert present_good.pk in pks
     assert future_good.pk in pks
+
+
+def test_commodity_measures(valid_user_client):
```
Comment (a-gleeson) [inline commodities/tests/test_views.py:188]: Should we maybe have a docstring for the tests?
These tests look great! 
The tests seem to follow a similar `setup`, `execution` and `validation pattern`, could this possibly be parameterised and condensed?

File: commodities/tests/test_views.py [hunk 3]
```diff
@@ -183,3 +183,152 @@ def test_goods_nomenclature(valid_user_client, date_ranges):
     assert past_good.pk not in pks
     assert present_good.pk in pks
     assert future_good.pk in pks
+
```
Comment (a-gleeson) [inline commodities/tests/test_views.py:186]: Should we include a test for a commodity with no measures?
And what happens when there are lots of measures? Are the results paginated?

File: commodities/jinja2/includes/commodities/tabs/measures.jinja (hunk 1 above)
Comment (eadpearce) [reply commodities/jinja2/includes/commodities/tabs/measures.jinja:5]: Oh! Probably not. Might be a holdover from a copypaste

File: commodities/tests/test_views.py (hunk 2 above)
Comment (eadpearce) [reply commodities/tests/test_views.py:188]: True yeah. I could see if I can make it into a fixture

File: commodities/tests/test_views.py (hunk 3 above)
Comment (eadpearce) [reply commodities/tests/test_views.py:186]: I'll ask Steve about pagination. I think in the tab layout it might be tricky to set up but I'll investigate

Comment (SteveCN7) [reply commodities/tests/test_views.py:186]: I think pagination is a nice to have, could we also consider a filter alternative?
Show the first twenty but allow a simplish filter by measure type perhaps?

**Review by a-gleeson — APPROVED** [review]
LGTM!

**Pull request merged**